src/core/Track.cpp

```diff
@@ -98,35 +89,36 @@ Track::~Track()
  *  \param tt The type of track to create
  *  \param tc The track container to attach to
  */
-Track * Track::create( Type tt, TrackContainer * tc )
+Track* Track::create(Type type, TrackContainer* trackContainer)
 {
-	Engine::audioEngine()->requestChangeInModel();
-
-	Track * t = nullptr;
-
-	switch( tt )
+	auto track = static_cast<Track*>(nullptr);
+	switch (type)
 	{
-		case Type::Instrument: t = new class InstrumentTrack( tc ); break;
-		case Type::Pattern: t = new class PatternTrack( tc ); break;
-		case Type::Sample: t = new class SampleTrack( tc ); break;
-//		case Type::Event:
-//		case Type::Video:
-		case Type::Automation: t = new class AutomationTrack( tc ); break;
-		case Type::HiddenAutomation:
-						t = new class AutomationTrack( tc, true ); break;
-		default: break;
+	case Type::Instrument:
+		track = new InstrumentTrack(trackContainer);
+		break;
+	case Type::Pattern:
+		track = new PatternTrack(trackContainer);
+		break;
+	case Type::Sample:
+		track = new SampleTrack(trackContainer);
+		break;
+	case Type::Automation:
```

Comment:
Wouldn't this set automation as hidden even if the automation is a regular one?

Reply:
I am using [fallthrough](https://en.cppreference.com/w/cpp/language/switch) here. In the case where the type of track is `Automation` or `HiddenAutomation`, we check if the type is `HiddenAutomation`. If it is, then the `hidden` variable inside the `AutomationTrack` constructor gives us a value of `true` and `false` otherwise.

I think I make this clearer though by using the `[[fallthrough]]` attribute.

Reply:
Ahh now i get the logic. You are checking within the AutomationTrack constructor, which i missed.